### PR TITLE
Customise chart colours and theme bug fixes

### PR DESCRIPTION
**Pull request by DanCunnington**
- ui_base.html tidy up
- fix for compass gauge pointer colour #137 
- added the ability to customise chart colours #137
- fix for less runtime bug #137

### CODE REVIEW

File: src/components/ui-gauge/ui-gauge.js [hunk 1]
```diff
@@ -9,11 +9,8 @@ angular.module('ui').directive('uiGauge', [ '$timeout', '$interpolate',
             link: function(scope, element, attrs) {
                 $timeout(function() {
                     var gauge;
-                    var bgnd = $('#toolbar').css("background-color");
-                    //Detect panel background colour and make foreground text b/w depending.
-                    var colors = $('ui-card-panel').css("background-color").match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
-                    var level = ((colors[1]*299) + (colors[2]*587) + (colors[3]*114))/1000;
-                    var fgnd = (level >= 128) ? '#111' : '#eee';
+                    var bgnd = scope.$eval('me.item.waveoptions').waveColor['theme-custom'];
```
Comment (dceejay) [inline src/components/ui-gauge/ui-gauge.js]: Why are you setting the bgnd color  to be the waveColor ? - if anything it should be the default colour or whatever the toolbar is... (which may be the wavecolor but may not be at some point) - or is that the only way to get it at this point ?

But yes should no longer need the calculation in here.

File: nodes/ui_chart.html [hunk 2]
```diff
@@ -169,6 +185,22 @@
         <input type="text" id="node-input-cutout" style="width:35px"> %
         </span>
     </div>
+    <div class="form-row" id="ui-chart-colours">
```
Comment (dceejay) [inline nodes/ui_chart.html]: So what happens after 9 ? (not a problem - just want to know...)

File: src/components/ui-gauge/ui-gauge.js (hunk 1 above)
Comment (DanCunnington) [inline src/components/ui-gauge/ui-gauge.js]: Ok fair enough - waveColor is currently the same as the base colour and bgnd is only ever used to set the wave or gauge colour, but I agree and will update bgnd to use the base. Cheers

File: nodes/ui_base.html [hunk 3]
```diff
@@ -564,7 +564,6 @@
                 addCustomisableStyle('group-textColor', 'Group Text', groupStyles);
                 addCustomisableStyle('group-borderColor', 'Group Border', groupStyles);
 
-
```
Comment (DanCunnington) [inline nodes/ui_base.html]: remove unnecessary whitespace

File: nodes/ui_chart.html (hunk 2 above)
Comment (DanCunnington) [inline nodes/ui_chart.html]: chart.js generates automatically. 9 fits nicely in the node panel and matches with the 3 per row as with gauges. Happy to adjust if you like :)

File: src/components/ui-gauge/ui-gauge.js (hunk 1 above)
Comment (DanCunnington) [inline src/components/ui-gauge/ui-gauge.js]: fixed in 1790172ad5558e172347ed3dc7879f21196e08ad thanks
